x/leverage/client/tests/util.go

```diff
@@ -132,7 +133,68 @@ func (s *IntegrationTestSuite) UpdateRegistry(
 			return prop.Status == govtypes.StatusPassed
 		},
 		2*time.Minute,
-		time.Second,
+		10*time.Second,
 		"proposal %d (%s) failed to pass", proposalID, content.Title,
 	)
 }
+
+// updateCollateralWeight modifies the collateral weight of a registered token identified by baseDenom.
+// Also returns its previous collateral weight, which is useful when undoing changes.
+func updateCollateralWeight(s *IntegrationTestSuite, baseDenom string, collateralWeight sdk.Dec) sdk.Dec {
+	val := s.network.Validators[0]
+	clientCtx := s.network.Validators[0].ClientCtx
+
+	queryFlags := []string{
+		fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+	}
+
+	// Query all registered tokens
+	out, err := clitestutil.ExecTestCLICmd(clientCtx, cli.GetCmdQueryAllRegisteredTokens(), queryFlags)
+	s.Require().NoError(err)
+
+	resp := &types.QueryRegisteredTokensResponse{}
+	s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), resp), out.String())
+
+	// Replace the collateral weight of the selected token with the new value
+	newTokens := resp.GetRegistry()
+	oldCollateralWeight := sdk.ZeroDec()
+	for _, token := range newTokens {
+		if token.BaseDenom == baseDenom {
+			oldCollateralWeight = token.CollateralWeight
+			token.CollateralWeight = collateralWeight
+		}
+	}
+
+	// Update token registry using the modified token registry - waits for proposal accepted
+	s.UpdateRegistry(
+		clientCtx,
+		types.NewUpdateRegistryProposal(
+			"test title",
+			"test description",
+			newTokens,
+		),
+		sdk.NewCoins(sdk.NewCoin(app.BondDenom, govtypes.DefaultMinDepositTokens)),
+		[]string{
+			fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+			fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+			fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+			fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(10))).String()),
+		}...,
+	)
+
+	// Query registered tokens again
+	out, err = clitestutil.ExecTestCLICmd(clientCtx, cli.GetCmdQueryAllRegisteredTokens(), queryFlags)
+	s.Require().NoError(err)
+
+	// Verify that the RegisteredTokens query is returning an updated collateral weight
+	registry := &types.QueryRegisteredTokensResponse{}
+	s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), registry), out.String())
+	for _, token := range registry.GetRegistry() {
+		if token.BaseDenom == baseDenom {
+			s.Require().Equal(collateralWeight, token.CollateralWeight, "Collateral weight not updated")
+		}
+	}
```

Comment:
@alexanderbez I'm currently blocked at this behavior: After the `UpdateRegistry` utility function is called to update collateral weight, further `QueryRegisteredToken` calls continue to return the old `token.CollateralWeight` instead of the updated one. The current code checks this once but I've also tried a `Require().Eventually` of 2 minutes to wait for the right collateral weight to be returned, to no success. In both cases, the liquidation tests fail in a way consistent with the collateral weight not having changed.

Since the `UpdateRegistry` function itself isn't failing, the proposal is definitely being accepted. Do you know what could be preventing it from being applied to the Token Registry?

Reply:
Hey @toteki I'm looking into this right now.

Reply:
So I just gave this a shot and it seems to work just fine.

https://pastebin.com/9v252N9Q

Notice I changed the collateral weight. It was 0.05 before and then 0.07 after. I queried before and after and you can see the token registry changing. So I'm not sure what issues you're running into.

Reply:
That's odd - the error I'm getting matches the failing `test-unit-cover` output on this PR. I'll look over vs. yours to see if I missed some detail.

So all of `make test-unit` succeeded on your machine when you tested?

Reply:
I didn't run any tests. I just made sure `UpdateRegistry` updated the token registry as it is intended.

Reply:
Nvm, I think I found the issue. I'll push the solution if it works

Reply:
Is the issue that tests are failing or the token registry is not being updated?

Reply:
Nice, what was the issue??

Reply:
The one I was stuck on seems to be modifying a copy of the registry then passing the original into UpdateRegistry, leading to no changes. There were other fails after that bug so I'm working on them now.